src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy

```diff
@@ -361,4 +361,30 @@ class PalantirDockerPluginTests extends AbstractPluginTest {
         execCond("docker rmi -f ${id}") || true
     }
 
+    def 'base image is pulled when "pull" parameter is set'() {
+        given:
+        String id = 'id8'
+        temporaryFolder.newFile('Dockerfile') << '''
+            FROM alpine:3.2
+        '''.stripIndent()
+        buildFile << """
+            plugins {
+                id 'com.palantir.docker'
+            }
+
+            docker {
+                name '${id}'
+                pull true
+            }
+        """.stripIndent()
+
+        when:
+        BuildResult buildResult = with('docker').build()
+
+        then:
+        buildResult.task(':docker').outcome == TaskOutcome.SUCCESS
+        buildResult.output.contains 'Pulling from library/alpine'
```

Comment:
this assertion looks bogus: it's true with `pull true` as well as `pull false` if `alpine:3.2` does not exist (which should be the case on a vanilla CI node.